Userland/Libraries/LibCore/MappedFile.cpp

```diff
@@ -14,35 +15,54 @@
 
 namespace Core {
 
-ErrorOr<NonnullRefPtr<MappedFile>> MappedFile::map(StringView path)
+ErrorOr<NonnullOwnPtr<MappedFile>> MappedFile::map(StringView path, OpenMode mode)
 {
-    auto fd = TRY(Core::System::open(path, O_RDONLY | O_CLOEXEC, 0));
-    return map_from_fd_and_close(fd, path);
+    auto file_mode = O_RDONLY;
+    if (mode == OpenMode::ReadWrite)
+        file_mode |= O_WRONLY;
```

Comment:
From the Linux version of the `open` manpage:

>    Unlike the other values that can be specified in flags, the
       access mode values O_RDONLY, O_WRONLY, and O_RDWR do not specify
       individual bits.  Rather, they define the low order two bits of
       flags, and are defined respectively as 0, 1, and 2.  In other
       words, the combination O_RDONLY | O_WRONLY is a logical error,
       and certainly does not have the same meaning as O_RDWR.

We just so happen to be doing that on Serenity, but that's not portable.